pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -0,0 +1,164 @@
+import logging
+import os
+import re
+import shutil
+from collections import namedtuple
+from datetime import datetime
+from pathlib import Path
+import traceback
+from typing import Dict, Iterable, List, Optional, Tuple
+
+import click
+import yaml
+from doozerlib.assembly import AssemblyTypes
+from pyartcd import constants, exectools
+from pyartcd.cli import cli, click_coroutine, pass_runtime
+from pyartcd.record import parse_record_log
+from pyartcd.runtime import Runtime
+from pyartcd.util import (get_assembly_type, get_release_name,
+                          isolate_el_version_in_release, load_group_config, load_releases_config)
+from semver import VersionInfo
+
+PlashetBuildResult = namedtuple("PlashetBuildResult", ("repo_name", "local_dir", "remote_url"))
+
+
+class BuildMicroShiftPipeline:
+    """ Rebase and build MicroShift for an assembly """
+
+    def __init__(self, runtime: Runtime, group: str, assembly: str,
+                 ocp_build_data_url: str, logger: Optional[logging.Logger] = None):
+        self.runtime = runtime
+        self.group = group
+        self.assembly = assembly
+        self._logger = logger or runtime.logger
+        self._slack_client = self.runtime.new_slack_client()
+
+        # determines OCP version
+        match = re.fullmatch(r"openshift-(\d+).(\d+)", group)
+        if not match:
+            raise ValueError(f"Invalid group name: {group}")
+        self._ocp_version = (int(match[1]), int(match[2]))
+
+        # sets environment variables for Doozer
+        self._doozer_env_vars = os.environ.copy()
+        self._doozer_env_vars["DOOZER_WORKING_DIR"] = str(self.runtime.working_dir / "doozer-working")
+
+        if not ocp_build_data_url:
+            ocp_build_data_url = self.runtime.config.get("build_config", {}).get("ocp_build_data_url")
+        if ocp_build_data_url:
+            self._doozer_env_vars["DOOZER_DATA_PATH"] = ocp_build_data_url
+
+    async def run(self):
+        self._slack_client.bind_channel(self.group)
+        slack_response = await self._slack_client.say(f":construction: Build microshift for assembly {self.assembly} :construction:")
+        slack_thread = slack_response["message"]["ts"]
+
+        try:
+            await load_group_config(self.group, self.assembly, env=self._doozer_env_vars)
+            releases_config = await load_releases_config(Path(self._doozer_env_vars["DOOZER_WORKING_DIR"], "ocp-build-data"))
+            assembly_type = get_assembly_type(releases_config, self.assembly)
+            if assembly_type not in [AssemblyTypes.STANDARD, AssemblyTypes.CANDIDATE, AssemblyTypes.PREVIEW]:
```

Comment:
What do you think about moving the allowed assembly types definition to the top of the file? This might help should we want to add a new type in the future

Reply:
I've added a static field `SUPPORTED_ASSEMBLY_TYPES` for this.